packages/kbn-i18n-react/src/provider.tsx

```diff
@@ -7,27 +7,36 @@
  */
 
 import React from 'react';
-
-// eslint-disable-next-line @kbn/eslint/module_migration
-import { IntlProvider } from 'react-intl';
-
 import { i18n } from '@kbn/i18n';
-import { PseudoLocaleWrapper } from './pseudo_locale_wrapper';
+import { IntlProvider } from 'react-intl';
 
 /**
  * The library uses the provider pattern to scope an i18n context to a tree
  * of components. This component is used to setup the i18n context for a tree.
  * IntlProvider should wrap react app's root component (inside each react render method).
  */
+export const I18nProvider = ({
+  children,
+}: {
+  children: IntlProvider['props']['children'] | React.ReactElement;
+}) => {
+  const { messages, formats, locale, defaultLocale, defaultFormats } = i18n.getTranslation();
```

Comment:
NIT I think it'd be cleaner if we check `isInitialized` before getting the translation properties.